src/client/static/analytics/ophan.ts

```diff
@@ -0,0 +1,56 @@
+import 'ophan-tracker-js';
+
+// the 'guardian.ophan' object is added by ophan-tracker-js
+// we extend the Window interface here to keep Typescript happy
+interface OphanWindow extends Window {
+  guardian: {
+    ophan: {
+      setEventEmitter: () => void;
+      trackComponentAttention: (
+        name: string,
+        el: Element,
+        visiblityThreshold: number,
+      ) => void;
+      record: ({}) => void;
+      viewId: string;
+      pageViewId: string;
+    };
+  };
+}
+
+declare let window: OphanWindow;
+
+interface OphanEvent {
+  experiences: string;
+  abTestRegister?: { [testId: string]: TestData };
+}
+
+interface TestData {
+  variantName: string;
+  complete: boolean;
+  campaignCodes?: Set<string>;
+}
+
+const record = (event: OphanEvent) => {
+  if (
+    window.guardian &&
```

Comment:
having called `import 'ophan-tracker-js';` I think these are set synchronously so we shouldn't have to test if record exists 🤔

Reply:
ah I see this is done in dotcom rendering. Maybe this is what we're supposed to do.
We shouldn't throw exceptions since failure to init ophan shouldn't prevent the page loading.

I don't know whether it's possible for `record` to throw exceptions, which we'd need to recover from

Reply:
I checked this locally by removing the import line. An error is thrown in the console, but it does not prevent the page from loading so it should not affect the user

Reply:
an exception would prevent the JavaScript after `ophanInit();`from being executed

Reply:
I take your point. If the Error is not adding any value, I think it can be removed